include/roscco/ros_to_oscc.h

```diff
@@ -13,6 +13,52 @@ extern "C" {
 #include <roscco/EnableDisable.h>
 #include <roscco/SteeringCommand.h>
 #include <roscco/ThrottleCommand.h>
+#include <std_msgs/Bool.h>
+
+// struct to enable detection of a rising or falling e_stop flag.
+struct e_stop_status {
+    bool current_e_stop_message{false};
+    bool previous_e_stop_message{false};
+    bool e_stop{false};
+
+    void setPrevious() {
+        previous_e_stop_message = current_e_stop_message;
+    }
+
+    bool isRising() {
+        if((!previous_e_stop_message) && current_e_stop_message) {
```

Comment:
We can reduce branching by simply returning the value, `return (!previous_e_stop_message) && current_e_stop_message;`

Reply:
Thanks, you're right, that would be a lot neater.